tools/NewModuleScript/src/main/kotlin/NewModuleScript.kt

```diff
@@ -0,0 +1,327 @@
+    // Create the sample resource folders
+    val destinationResDirectory = File("$samplesRepoPath/samples/$sampleWithHyphen")
+    destinationResDirectory.mkdirs()
+    // Display Map's res directory to copy over to new sample
+    val sourceResDirectory = File("$samplesRepoPath/samples/display-composable-mapview/")
+
+    // Screenshot image copied from source
+    val image = File("$samplesRepoPath/samples/$sampleWithHyphen/display-composable-mapview.png")
+
+    // Perform copy of the Android res folders from display-composable-mapview sample.
+    try {
+        FileUtils.copyDirectory(sourceResDirectory, destinationResDirectory)
+    } catch (e: IOException) {
+        e.printStackTrace()
+        exitProgram(e)
+    }
+
+    // Create the sample package directory in the source folder
+    val packageDirectory =
+        File("$samplesRepoPath/samples/$sampleWithHyphen/src/main/java/com/esri/arcgismaps/sample/$sampleWithoutSpaces")
+    if (!packageDirectory.exists()) {
+        packageDirectory.mkdirs()
+    } else {
+        exitProgram(Exception("Sample folder already exists!: ${packageDirectory.path}"))
+    }
+
+    // Copy Kotlin template files to new sample
+    val mainActivityTemplate = File("$samplesRepoPath/tools/NewModuleScript/MainActivityTemplate.kt")
+    val mapViewModelTemplate = File("$samplesRepoPath/tools/NewModuleScript/MapViewModelTemplate.kt")
+    val mainScreenTemplate = File("$samplesRepoPath/tools/NewModuleScript/MainScreenTemplate.kt")
+
+    // Perform copy
+    try {
+        FileUtils.copyFileToDirectory(mainActivityTemplate, packageDirectory)
+        var source = Paths.get("$packageDirectory/MainActivityTemplate.kt")
+        Files.move(source, source.resolveSibling("MainActivity.kt"))
+
+        var componentsDir = File("$packageDirectory/components")
+        componentsDir.mkdirs()
+
+        // copy and rename view-model
+        FileUtils.copyFileToDirectory(mapViewModelTemplate, componentsDir)
+        source = Paths.get("$componentsDir/MapViewModelTemplate.kt")
+        Files.move(source, source.resolveSibling("${sampleNameCamelCase}ViewModel.kt"))
+
+        // rename screenshot
+        Files.move(image.toPath(), image.toPath().resolveSibling("${sampleWithHyphen}.png"))
+
+        // copy and rename screens
+        componentsDir = File("$packageDirectory/screens")
+        componentsDir.mkdirs()
+        FileUtils.copyFileToDirectory(mainScreenTemplate, componentsDir)
+        source = Paths.get("$componentsDir/MainScreenTemplate.kt")
+        Files.move(source, source.resolveSibling("${sampleNameCamelCase}Screen.kt"))
+    } catch (e: IOException) {
+        e.printStackTrace()
+        exitProgram(e)
+    }
+}
+
+/**
+ * Exits the program with error -1 if it encounters an error
+ * @param e Error message to display
+ */
+private fun exitProgram(e: Exception) {
+    println("Error creating the sample: " + e.message)
+    println("StackTrace:")
+    e.printStackTrace()
```

Comment:
You might consider not printing the stacktrace, or passing a default boolean parameter to print it. It seems like most usage of this function is preceded by printing the stacktrace.